src/main/scala/viper/gobra/frontend/info/implementation/resolution/MemberResolution.scala

```diff
@@ -241,18 +248,65 @@ trait MemberResolution { this: TypeInfoImpl =>
   lazy val tryUnqualifiedPackageLookup: PIdnUse => Entity =
     attr[PIdnUse, Entity] {
       id => {
-        val unqualifiedImports = tree.root.imports.collect { case ui: PUnqualifiedImport => ui }
-        val results = unqualifiedImports.map(ui => tryPackageLookup(ui, id)).collect { case Some(r) => r }
-        if (results.isEmpty) {
-          UnknownEntity()
-        } else if (results.length > 1) {
-          MultipleEntity()
-        } else {
-          results.head._1
+        val transitiveParent = transitiveClosure(id, tree.parent(_))
+        val entities = for {
+          // get enclosing PProgram for this PIdnUse node
+          program <- transitiveParent(id).collectFirst { case p: PProgram => p }
+          // consider all unqualified imports for this program (not package)
+          unqualifiedImports = program.imports.collect { case ui: PUnqualifiedImport => ui }
+          // perform a package lookup in each unqualifiedly imported package
+          results = unqualifiedImports.map(ui => tryPackageLookup(ui, id)).collect { case Some(r) => r }
+        } yield results
+        entities match {
+          case Some(Vector(elem)) => elem._1
+          case Some(v) if v.length > 1 => MultipleEntity()
+          case _ => UnknownEntity()
         }
       }
     }
 
+  def transitiveClosure[T](t : T, onestep : T => Vector[T]): Relation[T, T] = {
+
+    def alternativeTransitiveClosure(t : T, onestep : T => Vector[T]): Relation[T, T] = {
+      val relation = new Relation[T, T]
+
+      @tailrec
+      def loop(pending : Queue[T]) : Relation[T, T] =
+        if (pending.isEmpty)
+          relation
+        else {
+          val l = pending.front
+          val next = onestep(l)
+          if (next.nonEmpty)
+            relation.putAll(t, next)
+          loop(pending.tail.enqueue(next))
+        }
+
+      loop(Queue(t))
+    }
+
+    def relationEquality(relation1: Relation[T, T], relation2: Relation[T, T]): Boolean = {
+      (relation1.pairs.forall(pair =>
+        relation2.containsInDomain(pair._1) &&
+          relation2(pair._1).contains(pair._2))
+      && relation2.pairs.forall(pair =>
+        relation1.containsInDomain(pair._1) &&
+          relation1(pair._1).contains(pair._2)))
```

Comment:
Can you not do something like:
`relation1.pairs.size == relation2.pairs.size && relation1.pairs containsAll relation2.pairs`

Reply:
Or can there be duplicates?

Reply:
won't fix as I'll remove this equality after deciding on the implementation